theano/gpuarray/subtensor.py

```diff
@@ -481,24 +481,46 @@ def c_code_cache_version(self):
         return (0,)
 
 
-class GpuAdvancedSubtensor(HideC, tensor.AdvancedSubtensor):
+def check_and_convert_boolean_masks(input, idx_list):
```

Comment:
The checks are done at `make_node` time, not at runtime in `perform`, so there is only one implementation. I thought it wouldn't do much harm to have the conversion in the `perform` for the non-bool op as well, because it just doesn't change anything and having a one-size-fits-all method keeps things simple.